### ISSUE TITLE
Logging deadlock issue - macOS app built using Xcode 9.2

### ISSUE THREAD

**Issue by d-metzler** [CocoaLumberjack/CocoaLumberjack#910]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [x] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | macOS
 Platform Version        | 10.12 Sierra
 CocoaLumberjack Version | 2.2.0
 Integration Method      | cocoapods
 Xcode Version           | Xcode 9.2
 Repro rate              | All the time (100%)

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

We have been using CocoaLumberjack v2.2.0 in our macOS application for over a year without any issues.  We recently upgrade from Xcode 8.3.3 to Xcode 9.2.  After rebuilding the application and debugging, CocoaLumberjack is now dead locking.  There have been no changes to the source files or cocoapod versions being used - the only difference is upgrading to Xcode 9.2

The application is using multiple threads and all logging is done asynchronously through DDLogInfo().

<img width="340" alt="screen shot 2018-01-23 at 8 46 19 am" src="https://user-images.githubusercontent.com/5521273/35285186-f6ab1aea-0019-11e8-99c0-17685c28fe7a.png">

<img width="340" alt="screen shot 2018-01-23 at 8 46 35 am" src="https://user-images.githubusercontent.com/5521273/35285199-fdf602b0-0019-11e8-95b8-09578d9774b5.png">

**Comment by bpoplauschi** [comment]
@d-metzler thanks for your interest in our project and reporting this issue back to us.
Can you check if #911 which was released with version 3.4.1 resolves the issue for you?
I see you are using version 2.2 of Lumberjack, I strongly suggest you upgrade to our latest version.
There is a migration guide available for you (since you are doing a major upgrade from 2.x to 3.x).

**Comment by d-metzler** [comment]
Using 3.4.2 the deadlock issue was still present.

**Comment by bpoplauschi** [comment]
Thanks for the quick response. I will look into this issue.
In the meantime, if you want to help, can you create a demo project where it's easy to repro this problem?

**Comment by d-metzler** [comment]
I will see if I can do that.

**Comment by ffried** [comment]
@d-metzler @bpoplauschi A demo project would be really helpful. I tried to reproduce this, but couldn't get it to deadlock. Looking at the code, I find it strange that it deadlocks like this. The calls seen here and in #911 are clearly all async calls and yet they get stuck in the `dispatch_async` call somehow.

**Comment by ffried** [comment]
I don't think so. In #507 the problem seems to be synchronous logging. Here it's definitively async.

**Comment by lolgear** [comment]
@ffried 
I only see `[SystemLogger info:]` in the MainThread.
Is it from `CocoaLumberjack`?
it seems that this logger ( if it custom ) could be related somehow to issue with custom loggers.

**Comment by ffried** [comment]
Nope, it's not from CocoaLumberjack. And yes it could be a custom logger. More likely just a wrapper around the macros like `DDLogInfo`, though, which is the same pattern as those of the comments in #911. Also, if it were synchronous, the dispatch_-frames would look differently.

**Comment by hhanesand** [comment]
@d-metzler Are you still having this issue? If you're able to reproduce this on the master branch, we'd love a full backtrace (via `bt all` in the debugger). Without this, there's not much we can do at the moment.

**Comment by lolgear** [comment]
@hhanesand I suggest to pin this issue.

**Comment by ffried** [comment]
@lolgear Why? It's not of particular interest to the community. We're just waiting on feedback here.
Pinned issues are good if it's something that won't be fixed and/or has an important workaround. Or a solution to a problem that's not a bug, but appears to often be seen as one.

**Comment by lolgear** [comment]
@ffried 
For protection from ubiquitous and hungry stale-bot.
It similar to Apple policy when they close bugs with "Attention" tag if there is no response, for example, in a month or two. Most of these bugs are system-wide and can't be reproduced by a click, heh.

It is ok if stale-bot can skip Awaiting response tag.

**Comment by ffried** [comment]
@lolgear stale-bot already leaves this alone thanks to the Important label 😉

**Comment by lolgear** [comment]
@stalebot good boy :3

@ffried and what about #941? It is so Important that bot marks it Stale.
Maybe it should take other form as page on site?

**Comment by ffried** [comment]
@lolgear I'd say the project is revived. 🙂 But yea, I've pinned that issue, since that makes sense to pin.

**Comment by bafnaparag** [comment]
Using latest version , Deadlock issue is still reproducible.
Everything works fine if I use LOG_ASYNC_ENABLED for error logs.

**Comment by hhanesand** [comment]
@bafnaparag Like I said above, we're not going to be able to help you unless you post a full backtrace...

Also, what do you mean by "latest version"?

**Comment by bafnaparag** [comment]
@hhanesand So I am getting issue even after using LOG_ASYNC_ENABLED. Here is sample

(lldb) bt all
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e1071f libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 111
    frame #3: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #4: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
  * frame #5: 0x000000010059d2b4 MyApplication`-[DDLog queueLogMessage:asynchronously:](self=0x000060000000c390, _cmd="queueLogMessage:asynchronously:", logMessage=0x0000600002c96e00, asyncFlag=YES) at DDLog.m:362
    frame #6: 0x000000010059e1cd MyApplication`-[DDLog log:message:level:flag:context:file:function:line:tag:](self=0x000060000000c390, _cmd="log:message:level:flag:context:file:function:line:tag:", asynchronous=YES, message=@"PBServer authenticated", level=DDLogLevelDebug, flag=DDLogFlagInfo, context=0, file="/Users/Parag/Developer/MyProject/Common/PBModel.m", function="-[PBModel createAndStartAccountWith:credential:completionBlock:]", line=735, tag=0x0000000000000000) at DDLog.m:508
    frame #7: 0x000000010059df2b MyApplication`+[DDLog log:message:level:flag:context:file:function:line:tag:](self=DDLog, _cmd="log:message:level:flag:context:file:function:line:tag:", asynchronous=YES, message=@"PBServer authenticated", level=DDLogLevelDebug, flag=DDLogFlagInfo, context=0, file="/Users/Parag/Developer/MyProject/Common/PBModel.m", function="-[PBModel createAndStartAccountWith:credential:completionBlock:]", line=735, tag=0x0000000000000000) at DDLog.m:485
    frame #8: 0x000000010059d73b MyApplication`+[DDLog log:level:flag:context:file:function:line:tag:format:](self=DDLog, _cmd="log:level:flag:context:file:function:line:tag:format:", asynchronous=YES, level=DDLogLevelDebug, flag=DDLogFlagInfo, context=0, file="/Users/Parag/Developer/MyProject/Common/PBModel.m", function="-[PBModel createAndStartAccountWith:credential:completionBlock:]", line=735, tag=0x0000000000000000, format=@"PBServer authenticated") at DDLog.m:391
    frame #9: 0x00000001003ad90d MyApplication`-[PBModel createAndStartAccountWith:credential:completionBlock:](self=0x0000600002112a30, _cmd="createAndStartAccountWith:credential:completionBlock:", server=0x0000600002927790, credential=0x000060000032e660, completionBlock=0x00000001002ef260) at PBModel.m:735
    frame #10: 0x00000001002ef1d8 MyApplication`-[TopLevelController didAuthenticate:credentia:](self=0x0000600002112120, _cmd="didAuthenticate:credentia:", server=0x0000600002927790, credential=0x000060000032e660) at TopLevelController.m:722
    frame #11: 0x00000001000a4ee2 MyApplication`__53-[PBXOAuth2ViewController didAuthenticateCredential:]_block_invoke(.block_descriptor=0x000070000b15ca58) at PBXOAuth2ViewController.m:225
    frame #12: 0x0000000101bac7c3 libdispatch.dylib`_dispatch_client_callout + 8
    frame #13: 0x0000000101bbc445 libdispatch.dylib`_dispatch_async_and_wait_invoke + 179
    frame #14: 0x0000000101bac7c3 libdispatch.dylib`_dispatch_client_callout + 8
    frame #15: 0x0000000101bbb390 libdispatch.dylib`_dispatch_main_queue_callback_4CF + 1909
    frame #16: 0x00007fff39a581ab CoreFoundation`__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
    frame #17: 0x00007fff39a578ba CoreFoundation`__CFRunLoopRun + 2335
    frame #18: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #19: 0x00007fff38cedab5 HIToolbox`RunCurrentEventLoopInMode + 293
    frame #20: 0x00007fff38ced7eb HIToolbox`ReceiveNextEventCommon + 618
    frame #21: 0x00007fff38ced568 HIToolbox`_BlockUntilNextEventMatchingListInModeWithFilter + 64
    frame #22: 0x00007fff36fa8363 AppKit`_DPSNextEvent + 997
    frame #23: 0x00007fff36fa7102 AppKit`-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1362
    frame #24: 0x00007fff36fa1165 AppKit`-[NSApplication run] + 699
    frame #25: 0x00007fff36f908a3 AppKit`NSApplicationMain + 780
    frame #26: 0x000000010000c8f2 MyApplication`main(argc=7, argv=0x00007ffeefbff4d0) at main.m:11
    frame #27: 0x00007fff66ca8ed9 libdyld.dylib`start + 1
  thread #7
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e1071f libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 111
    frame #3: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #4: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
    frame #5: 0x000000010059d2b4 MyApplication`-[DDLog queueLogMessage:asynchronously:](self=0x000060000000c390, _cmd="queueLogMessage:asynchronously:", logMessage=0x0000600002c58e80, asyncFlag=YES) at DDLog.m:362
    frame #6: 0x000000010059e1cd MyApplication`-[DDLog log:message:level:flag:context:file:function:line:tag:](self=0x000060000000c390, _cmd="log:message:level:flag:context:file:function:line:tag:", asynchronous=YES, message=@"[FinderSync] failed to send reply,mach error (ipc/send) invalid destination port", level=DDLogLevelDebug, flag=DDLogFlagWarning, context=0, file="/Users/Parag/Developer/MyProject/Mac/project/PBIPCWithFinder.m", function="void handleSessionAndOfflineStatus(mach_msg_command_rcv_t *)", line=207, tag=0x0000000000000000) at DDLog.m:508
    frame #7: 0x000000010059df2b MyApplication`+[DDLog log:message:level:flag:context:file:function:line:tag:](self=DDLog, _cmd="log:message:level:flag:context:file:function:line:tag:", asynchronous=YES, message=@"[FinderSync] failed to send reply,mach error (ipc/send) invalid destination port", level=DDLogLevelDebug, flag=DDLogFlagWarning, context=0, file="/Users/Parag/Developer/MyProject/Mac/project/PBIPCWithFinder.m", function="void handleSessionAndOfflineStatus(mach_msg_command_rcv_t *)", line=207, tag=0x0000000000000000) at DDLog.m:485
    frame #8: 0x000000010059d73b MyApplication`+[DDLog log:level:flag:context:file:function:line:tag:format:](self=DDLog, _cmd="log:level:flag:context:file:function:line:tag:format:", asynchronous=YES, level=DDLogLevelDebug, flag=DDLogFlagWarning, context=0, file="/Users/Parag/Developer/MyProject/Mac/project/PBIPCWithFinder.m", function="void handleSessionAndOfflineStatus(mach_msg_command_rcv_t *)", line=207, tag=0x0000000000000000, format=@"[FinderSync] failed to send reply,mach error %s") at DDLog.m:391
    frame #9: 0x000000010034c271 MyApplication`handleSessionAndOfflineStatus(msg=0x000070000b1ded70) at PBIPCWithFinder.m:207
    frame #10: 0x000000010034b92a MyApplication`-[PBIPCWithFinder handleMachMessage:](self=0x00006000000048c0, _cmd="handleMachMessage:", machMessage=0x000070000b1ded70) at PBIPCWithFinder.m:164
    frame #11: 0x00007fff3be07728 Foundation`__NSFireMachPort + 82
    frame #12: 0x00007fff39a6faef CoreFoundation`__CFMachPortPerform + 282
    frame #13: 0x00007fff39a6f9c9 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 41
    frame #14: 0x00007fff39a6f927 CoreFoundation`__CFRunLoopDoSource1 + 527
    frame #15: 0x00007fff39a579a9 CoreFoundation`__CFRunLoopRun + 2574
    frame #16: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #17: 0x00007fff3bdbd4aa Foundation`-[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 280
    frame #18: 0x000000010034b485 MyApplication`-[PBIPCWithFinder start](self=0x00006000000048c0, _cmd="start") at PBIPCWithFinder.m:122
    frame #19: 0x00007fff3bdb3104 Foundation`__NSThread__start__ + 1218
    frame #20: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #21: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #22: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #11, name = 'com.apple.NSEventThread'
    frame #0: 0x00007fff66de217a libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff66de26d0 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff39a580c2 CoreFoundation`__CFRunLoopServiceMachPort + 337
    frame #3: 0x00007fff39a57611 CoreFoundation`__CFRunLoopRun + 1654
    frame #4: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #5: 0x00007fff36fb0f89 AppKit`_NSEventThread + 160
    frame #6: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #7: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #8: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #14, name = 'com.apple.NSURLConnectionLoader'
    frame #0: 0x00007fff66de217a libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff66de26d0 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff39a580c2 CoreFoundation`__CFRunLoopServiceMachPort + 337
    frame #3: 0x00007fff39a57611 CoreFoundation`__CFRunLoopRun + 1654
    frame #4: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #5: 0x00007fff38957cc5 CFNetwork`-[__CoreSchedulingSetRunnable runForever] + 195
    frame #6: 0x00007fff3bdb3104 Foundation`__NSThread__start__ + 1218
    frame #7: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #8: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #9: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #20, queue = 'com.apple.root.default-qos'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e0fd74 libBacktraceRecording.dylib`get_entry_from_free_list + 38
    frame #3: 0x0000000100e10a9e libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 1006
    frame #4: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #5: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
    frame #6: 0x00007fff3892323b CFNetwork`QCoreSchedulingSet::performAsync(void () block_pointer) const + 85
    frame #7: 0x00007fff38ab1453 CFNetwork`___ZN19HTTPConnectionCache11enableTimerEv_block_invoke + 73
    frame #8: 0x0000000101bac7c3 libdispatch.dylib`_dispatch_client_callout + 8
    frame #9: 0x0000000101bafbcd libdispatch.dylib`_dispatch_continuation_pop + 576
    frame #10: 0x0000000101bc4506 libdispatch.dylib`_dispatch_source_invoke + 2109
    frame #11: 0x0000000101bbf270 libdispatch.dylib`_dispatch_root_queue_drain + 334
    frame #12: 0x0000000101bbfd33 libdispatch.dylib`_dispatch_worker_thread2 + 125
    frame #13: 0x0000000101c27119 libsystem_pthread.dylib`_pthread_wqthread + 619
    frame #14: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #22, name = 'SyncWatcher'
    frame #0: 0x00007fff66de217a libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff66de26d0 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff39a580c2 CoreFoundation`__CFRunLoopServiceMachPort + 337
    frame #3: 0x00007fff39a57611 CoreFoundation`__CFRunLoopRun + 1654
    frame #4: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #5: 0x00007fff39a56b54 CoreFoundation`CFRunLoopRun + 40
    frame #6: 0x00000001001dcc19 MyApplication`-[SyncWatcher runWatcherThread](self=0x0000600002121090, _cmd="runWatcherThread") at SyncWatcher.m:59
    frame #7: 0x00007fff3bdb3104 Foundation`__NSThread__start__ + 1218
    frame #8: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #9: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #10: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #23, name = 'PBEvents'
    frame #0: 0x00007fff66de217a libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff66de26d0 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff39a580c2 CoreFoundation`__CFRunLoopServiceMachPort + 337
    frame #3: 0x00007fff39a57611 CoreFoundation`__CFRunLoopRun + 1654
    frame #4: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #5: 0x00007fff39a56b54 CoreFoundation`CFRunLoopRun + 40
    frame #6: 0x000000010006a4e7 MyApplication`-[SCEvents runEventThread](self=0x000060000370da40, _cmd="runEventThread") at SCEvents.m:130
    frame #7: 0x00007fff3bdb3104 Foundation`__NSThread__start__ + 1218
    frame #8: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #9: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #10: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #29, name = 'JavaScriptCore bmalloc scavenger'
    frame #0: 0x00007fff66de57de libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x0000000101c2b1eb libsystem_pthread.dylib`_pthread_cond_wait + 724
    frame #2: 0x00007fff6437dbda libc++.1.dylib`std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
    frame #3: 0x00007fff3d098cb2 JavaScriptCore`void std::__1::condition_variable_any::wait<std::__1::unique_lock<bmalloc::Mutex> >(std::__1::unique_lock<bmalloc::Mutex>&) + 82
    frame #4: 0x00007fff3d09ce0b JavaScriptCore`bmalloc::Scavenger::threadRunLoop() + 139
    frame #5: 0x00007fff3d09c5e9 JavaScriptCore`bmalloc::Scavenger::threadEntryPoint(bmalloc::Scavenger*) + 9
    frame #6: 0x00007fff3d09e017 JavaScriptCore`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(bmalloc::Scavenger*), bmalloc::Scavenger*> >(void*) + 39
    frame #7: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #8: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #9: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #31, queue = 'NSOperationQueue 0x6000002312c0 (QOS: UNSPECIFIED)'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x0000000101bad0b6 libdispatch.dylib`_dispatch_ulock_wait + 47
    frame #2: 0x0000000101bad195 libdispatch.dylib`_dispatch_thread_event_wait_slow + 33
    frame #3: 0x0000000101bbc66d libdispatch.dylib`__DISPATCH_WAIT_FOR_QUEUE__ + 309
    frame #4: 0x0000000101bbbe3f libdispatch.dylib`_dispatch_sync_f_slow + 223
    frame #5: 0x000000010033b1ac MyApplication`runOnMainQueue(block=0x00000001000a4ce0) at PBMainThreadRun.m:18
    frame #6: 0x00000001000a4c9b MyApplication`-[PBXOAuth2ViewController didAuthenticateCredential:](self=0x0000600003516940, _cmd="didAuthenticateCredential:", credential=0x000060000032e660) at PBXOAuth2ViewController.m:216
    frame #7: 0x00000001005f1144 MyApplication`__57-[PBAuthenticationDriver(.block_descriptor=0x0000600000ce2dc0, data=0x0000600001877800, response=0x000060000022fba0, error=0x0000000000000000) processFrameLoadRequest:]_block_invoke at PBAuthenticationDriver+OAuth.m:118
    frame #8: 0x00007fff3892b523 CFNetwork`__75-[__NSURLSessionLocal taskForClass:request:uploadFile:bodyData:completion:]_block_invoke + 19
    frame #9: 0x00007fff3892af24 CFNetwork`__49-[__NSCFLocalSessionTask _task_onqueue_didFinish]_block_invoke + 172
    frame #10: 0x00007fff3bdf7917 Foundation`__NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
    frame #11: 0x00007fff3bdf781e Foundation`-[NSBlockOperation main] + 68
    frame #12: 0x00007fff3bdcce4a Foundation`-[__NSOperationInternal _start:] + 686
    frame #13: 0x00007fff3bdf754f Foundation`__NSOQSchedule_f + 227
    frame #14: 0x0000000101bab6eb libdispatch.dylib`_dispatch_call_block_and_release + 12
    frame #15: 0x0000000101bac7c3 libdispatch.dylib`_dispatch_client_callout + 8
    frame #16: 0x0000000101bafbcd libdispatch.dylib`_dispatch_continuation_pop + 576
    frame #17: 0x0000000101baeca3 libdispatch.dylib`_dispatch_async_redirect_invoke + 780
    frame #18: 0x0000000101bbf270 libdispatch.dylib`_dispatch_root_queue_drain + 334
    frame #19: 0x0000000101bbfd33 libdispatch.dylib`_dispatch_worker_thread2 + 125
    frame #20: 0x0000000101c27119 libsystem_pthread.dylib`_pthread_wqthread + 619
    frame #21: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #33, queue = 'cocoa.lumberjack'
    frame #0: 0x00007fff66de6a86 libsystem_kernel.dylib`__psynch_rw_wrlock + 10
    frame #1: 0x0000000101c299d2 libsystem_pthread.dylib`_pthread_rwlock_lock_wait + 89
    frame #2: 0x0000000101c2765b libsystem_pthread.dylib`_pthread_rwlock_lock_slow + 558
    frame #3: 0x0000000100e14739 libBacktraceRecording.dylib`remove_stack_from_backtrace_uniquing_table + 33
    frame #4: 0x0000000100e1147f libBacktraceRecording.dylib`decrement_work_item_refcount + 80
    frame #5: 0x0000000100e10d7b libBacktraceRecording.dylib`gcd_queue_item_complete_hook + 166
    frame #6: 0x0000000101bb49f1 libdispatch.dylib`_dispatch_lane_serial_drain + 831
    frame #7: 0x0000000101bb587e libdispatch.dylib`_dispatch_lane_invoke + 446
    frame #8: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #9: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #10: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #34, name = 'com.apple.coreanimation.render-server'
    frame #0: 0x00007fff66de217a libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff66de26d0 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff44a6cc74 QuartzCore`CA::Render::Server::server_thread(void*) + 956
    frame #3: 0x00007fff44a6c8a2 QuartzCore`thread_fun(void*) + 25
    frame #4: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #5: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #6: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #35, queue = 'com.apple.root.default-qos.overcommit'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e1071f libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 111
    frame #3: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #4: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
    frame #5: 0x00007fff3b0aa85d LaunchServices`LSNotificationReceiver::receiveNotificationFromServer(_xpc_connection_s*, void*) + 973
    frame #6: 0x00007fff3b0aa46c LaunchServices`___ZN22LSNotificationReceiver23getNotificationListenerEv_block_invoke.10 + 414
    frame #7: 0x00007fff66edca28 libxpc.dylib`_xpc_connection_call_event_handler + 56
    frame #8: 0x00007fff66eda9be libxpc.dylib`_xpc_connection_mach_event + 933
    frame #9: 0x0000000101bac863 libdispatch.dylib`_dispatch_client_callout4 + 9
    frame #10: 0x0000000101bc80a9 libdispatch.dylib`_dispatch_mach_msg_invoke + 487
    frame #11: 0x0000000101bb47c9 libdispatch.dylib`_dispatch_lane_serial_drain + 279
    frame #12: 0x0000000101bc93ed libdispatch.dylib`_dispatch_mach_invoke + 557
    frame #13: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #14: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #15: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
    frame #16: 0x0000000101c07600 libdispatch.dylib`_dispatch_root_queues + 896
  thread #37, queue = 'com.apple.network.connections'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e10bcc libBacktraceRecording.dylib`gcd_queue_item_dequeue_hook + 219
    frame #3: 0x0000000101be23c6 libdispatch.dylib`_dispatch_introspection_queue_item_dequeue_hook + 46
    frame #4: 0x0000000101bafb35 libdispatch.dylib`_dispatch_continuation_pop + 424
    frame #5: 0x0000000101bc4506 libdispatch.dylib`_dispatch_source_invoke + 2109
    frame #6: 0x0000000101bb73e7 libdispatch.dylib`_dispatch_workloop_invoke + 3562
    frame #7: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #8: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #9: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #38
    frame #0: 0x00007fff66de3b6a libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x0000000101c2706c libsystem_pthread.dylib`_pthread_wqthread + 446
    frame #2: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #39, queue = 'com.apple.root.default-qos'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e10bcc libBacktraceRecording.dylib`gcd_queue_item_dequeue_hook + 219
    frame #3: 0x0000000101be23c6 libdispatch.dylib`_dispatch_introspection_queue_item_dequeue_hook + 46
    frame #4: 0x0000000101bafb35 libdispatch.dylib`_dispatch_continuation_pop + 424
    frame #5: 0x0000000101bc4506 libdispatch.dylib`_dispatch_source_invoke + 2109
    frame #6: 0x0000000101bbf270 libdispatch.dylib`_dispatch_root_queue_drain + 334
    frame #7: 0x0000000101bbfd33 libdispatch.dylib`_dispatch_worker_thread2 + 125
    frame #8: 0x0000000101c27119 libsystem_pthread.dylib`_pthread_wqthread + 619
    frame #9: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #40
    frame #0: 0x00007fff66de3b6a libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x0000000101c2706c libsystem_pthread.dylib`_pthread_wqthread + 446
    frame #2: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #42
    frame #0: 0x00007fff66de3b6a libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x0000000101c2714c libsystem_pthread.dylib`_pthread_wqthread + 670
    frame #2: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #43
    frame #0: 0x0000000000000000

"latest version" : Update my code on Jan 8.

**Comment by bafnaparag** [comment]
@hhanesand I got deadlock with LOG_ASYNC_ENABLED.

Here is sample

(lldb) bt all
  thread #1, queue = 'com.apple.main-thread'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e1071f libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 111
    frame #3: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #4: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
    frame #5: 0x000000010059d1b4 MyApp`-[DDLog queueLogMessage:asynchronously:](self=0x0000600000008600, _cmd="queueLogMessage:asynchronously:", logMessage=0x0000600002c77200, asyncFlag=YES) at DDLog.m:362
    frame #6: 0x000000010059e0cd MyApp`-[DDLog log:message:level:flag:context:file:function:line:tag:](self=0x0000600000008600, _cmd="log:message:level:flag:context:file:function:line:tag:", asynchronous=YES, message=@"Request is nil", level=DDLogLevelDebug, flag=DDLogFlagDebug, context=0, file="/Users/UserName/Developer/PBApple/CoreSDK/MyAppAPI/MyAppAPI/PBDataTask.m", function="-[PBDataTask start]", line=154, tag=0x0000000000000000) at DDLog.m:508
    frame #7: 0x000000010059de2b MyApp`+[DDLog log:message:level:flag:context:file:function:line:tag:](self=DDLog, _cmd="log:message:level:flag:context:file:function:line:tag:", asynchronous=YES, message=@"Request is nil", level=DDLogLevelDebug, flag=DDLogFlagDebug, context=0, file="/Users/UserName/Developer/PBApple/CoreSDK/MyAppAPI/MyAppAPI/PBDataTask.m", function="-[PBDataTask start]", line=154, tag=0x0000000000000000) at DDLog.m:485
    frame #8: 0x000000010059d63b MyApp`+[DDLog log:level:flag:context:file:function:line:tag:format:](self=DDLog, _cmd="log:level:flag:context:file:function:line:tag:format:", asynchronous=YES, level=DDLogLevelDebug, flag=DDLogFlagDebug, context=0, file="/Users/UserName/Developer/PBApple/CoreSDK/MyAppAPI/MyAppAPI/PBDataTask.m", function="-[PBDataTask start]", line=154, tag=0x0000000000000000, format=@"Request is nil") at DDLog.m:391
    frame #9: 0x00000001005cff01 MyApp`-[PBDataTask start](self=0x0000600003d094a0, _cmd="start") at PBDataTask.m:154
    frame #10: 0x00000001005b4eae MyApp`-[PBUploadNePBocumentRequest start](self=0x0000600003d094a0, _cmd="start") at PBUploadNePBocumentRequest.m:124
    frame #11: 0x00000001001ff23d MyApp`-[PBQRunLoopOperationWithRequest operationDidStart](self=0x000060000303cab0, _cmd="operationDidStart") at PBQRunLoopOperationWithRequest.m:49
    frame #12: 0x000000010015b788 MyApp`-[UploadDocumentOperation operationDidStart](self=0x000060000303cab0, _cmd="operationDidStart") at UploadDocumentOperation.m:401
    frame #13: 0x00000001001cd7da MyApp`-[PBQRunLoopOperation startOnRunLoopThread](self=0x000060000303cab0, _cmd="startOnRunLoopThread") at PBQRunLoopOperation.m:241
    frame #14: 0x00007fff3be52bab Foundation`__NSThreadPerformPerform + 327
    frame #15: 0x00007fff39a74475 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
    frame #16: 0x00007fff39a7441b CoreFoundation`__CFRunLoopDoSource0 + 108
    frame #17: 0x00007fff39a57eb5 CoreFoundation`__CFRunLoopDoSources0 + 195
    frame #18: 0x00007fff39a5745e CoreFoundation`__CFRunLoopRun + 1219
    frame #19: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #20: 0x00007fff38cedab5 HIToolbox`RunCurrentEventLoopInMode + 293
    frame #21: 0x00007fff38ced7eb HIToolbox`ReceiveNextEventCommon + 618
    frame #22: 0x00007fff38ced568 HIToolbox`_BlockUntilNextEventMatchingListInModeWithFilter + 64
    frame #23: 0x00007fff36fa8363 AppKit`_DPSNextEvent + 997
    frame #24: 0x00007fff36fa7102 AppKit`-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1362
    frame #25: 0x00007fff36fa1165 AppKit`-[NSApplication run] + 699
    frame #26: 0x00007fff36f908a3 AppKit`NSApplicationMain + 780
    frame #27: 0x000000010000c172 MyApp`main(argc=7, argv=0x00007ffeefbff4d0) at main.m:11
    frame #28: 0x00007fff66ca8ed9 libdyld.dylib`start + 1
  thread #6
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e1071f libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 111
    frame #3: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #4: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
    frame #5: 0x000000010059d1b4 MyApp`-[DDLog queueLogMessage:asynchronously:](self=0x0000600000008600, _cmd="queueLogMessage:asynchronously:", logMessage=0x0000600002c77000, asyncFlag=YES) at DDLog.m:362
    frame #6: 0x000000010059e0cd MyApp`-[DDLog log:message:level:flag:context:file:function:line:tag:](self=0x0000600000008600, _cmd="log:message:level:flag:context:file:function:line:tag:", asynchronous=YES, message=@"[FinderSync] failed to send reply,mach error (ipc/send) invalid destination port", level=DDLogLevelDebug, flag=DDLogFlagWarning, context=0, file="/Users/UserName/Developer/PBApple/Mac/MacMyApp/PBIPCWithFinder.m", function="void handleSessionAndOfflineStatus(mach_msg_command_rcv_t *)", line=207, tag=0x0000000000000000) at DDLog.m:508
    frame #7: 0x000000010059de2b MyApp`+[DDLog log:message:level:flag:context:file:function:line:tag:](self=DDLog, _cmd="log:message:level:flag:context:file:function:line:tag:", asynchronous=YES, message=@"[FinderSync] failed to send reply,mach error (ipc/send) invalid destination port", level=DDLogLevelDebug, flag=DDLogFlagWarning, context=0, file="/Users/UserName/Developer/PBApple/Mac/MacMyApp/PBIPCWithFinder.m", function="void handleSessionAndOfflineStatus(mach_msg_command_rcv_t *)", line=207, tag=0x0000000000000000) at DDLog.m:485
    frame #8: 0x000000010059d63b MyApp`+[DDLog log:level:flag:context:file:function:line:tag:format:](self=DDLog, _cmd="log:level:flag:context:file:function:line:tag:format:", asynchronous=YES, level=DDLogLevelDebug, flag=DDLogFlagWarning, context=0, file="/Users/UserName/Developer/PBApple/Mac/MacMyApp/PBIPCWithFinder.m", function="void handleSessionAndOfflineStatus(mach_msg_command_rcv_t *)", line=207, tag=0x0000000000000000, format=@"[FinderSync] failed to send reply,mach error %s") at DDLog.m:391
    frame #9: 0x000000010034bbb1 MyApp`handleSessionAndOfflineStatus(msg=0x000070000a675d70) at PBIPCWithFinder.m:207
    frame #10: 0x000000010034b26a MyApp`-[PBIPCWithFinder handleMachMessage:](self=0x0000600000010e30, _cmd="handleMachMessage:", machMessage=0x000070000a675d70) at PBIPCWithFinder.m:164
    frame #11: 0x00007fff3be07728 Foundation`__NSFireMachPort + 82
    frame #12: 0x00007fff39a6faef CoreFoundation`__CFMachPortPerform + 282
    frame #13: 0x00007fff39a6f9c9 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 41
    frame #14: 0x00007fff39a6f927 CoreFoundation`__CFRunLoopDoSource1 + 527
    frame #15: 0x00007fff39a579a9 CoreFoundation`__CFRunLoopRun + 2574
    frame #16: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #17: 0x00007fff3bdbd4aa Foundation`-[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 280
    frame #18: 0x000000010034adc5 MyApp`-[PBIPCWithFinder start](self=0x0000600000010e30, _cmd="start") at PBIPCWithFinder.m:122
    frame #19: 0x00007fff3bdb3104 Foundation`__NSThread__start__ + 1218
    frame #20: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #21: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #22: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #12, name = 'com.apple.NSEventThread'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e1071f libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 111
    frame #3: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #4: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
    frame #5: 0x00007fff5fb66c97 SkyLight`CGSSnarfAndDispatchDatagrams + 3979
    frame #6: 0x00007fff5fdbc658 SkyLight`SLSGetNextEventRecordInternal + 81
    frame #7: 0x00007fff5fc6e1a2 SkyLight`SLEventCreateNextEvent + 137
    frame #8: 0x00007fff38cf79d2 HIToolbox`PullEventsFromWindowServerOnConnection(unsigned int, unsigned char, __CFMachPortBoost*) + 45
    frame #9: 0x00007fff38cf796d HIToolbox`MessageHandler(__CFMachPort*, void*, long, void*) + 51
    frame #10: 0x00007fff39a6facb CoreFoundation`__CFMachPortPerform + 246
    frame #11: 0x00007fff39a6f9c9 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 41
    frame #12: 0x00007fff39a6f927 CoreFoundation`__CFRunLoopDoSource1 + 527
    frame #13: 0x00007fff39a579a9 CoreFoundation`__CFRunLoopRun + 2574
    frame #14: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #15: 0x00007fff36fb0f89 AppKit`_NSEventThread + 160
    frame #16: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #17: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #18: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #13, name = 'SyncWatcher'
    frame #0: 0x00007fff66de217a libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff66de26d0 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff39a580c2 CoreFoundation`__CFRunLoopServiceMachPort + 337
    frame #3: 0x00007fff39a57611 CoreFoundation`__CFRunLoopRun + 1654
    frame #4: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #5: 0x00007fff39a56b54 CoreFoundation`CFRunLoopRun + 40
    frame #6: 0x00000001001dc5b9 MyApp`-[SyncWatcher runWatcherThread](self=0x000060000211da40, _cmd="runWatcherThread") at SyncWatcher.m:59
    frame #7: 0x00007fff3bdb3104 Foundation`__NSThread__start__ + 1218
    frame #8: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #9: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #10: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #14, name = 'PBEvents'
    frame #0: 0x00007fff66de217a libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff66de26d0 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff39a580c2 CoreFoundation`__CFRunLoopServiceMachPort + 337
    frame #3: 0x00007fff39a57611 CoreFoundation`__CFRunLoopRun + 1654
    frame #4: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #5: 0x00007fff39a56b54 CoreFoundation`CFRunLoopRun + 40
    frame #6: 0x0000000100069d67 MyApp`-[SCEvents runEventThread](self=0x0000600003709a40, _cmd="runEventThread") at SCEvents.m:130
    frame #7: 0x00007fff3bdb3104 Foundation`__NSThread__start__ + 1218
    frame #8: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #9: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #10: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
  thread #15, name = 'com.apple.NSURLConnectionLoader'
    frame #0: 0x00007fff66de217a libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff66de26d0 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff39a580c2 CoreFoundation`__CFRunLoopServiceMachPort + 337
    frame #3: 0x00007fff39a57611 CoreFoundation`__CFRunLoopRun + 1654
    frame #4: 0x00007fff39a56d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #5: 0x00007fff38957cc5 CFNetwork`-[__CoreSchedulingSetRunnable runForever] + 195
    frame #6: 0x00007fff3bdb3104 Foundation`__NSThread__start__ + 1218
    frame #7: 0x0000000101c27e35 libsystem_pthread.dylib`_pthread_body + 126
    frame #8: 0x0000000101c2aec7 libsystem_pthread.dylib`_pthread_start + 70
    frame #9: 0x0000000101c26e51 libsystem_pthread.dylib`thread_start + 13
* thread #202, queue = 'cocoa.lumberjack'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x0000000101bad0b6 libdispatch.dylib`_dispatch_ulock_wait + 47
    frame #2: 0x0000000101bad84d libdispatch.dylib`_dispatch_group_wait_slow + 40
  * frame #3: 0x00000001005a0d4b MyApp`-[DDLog lt_log:](self=0x0000600000008600, _cmd="lt_log:", logMessage=0x0000600002c76b80) at DDLog.m:844
    frame #4: 0x000000010059d293 MyApp`__40-[DDLog queueLogMessage:asynchronously:]_block_invoke(.block_descriptor=0x0000600000cf2340) at DDLog.m:357
    frame #5: 0x0000000101bab6eb libdispatch.dylib`_dispatch_call_block_and_release + 12
    frame #6: 0x0000000101bac7c3 libdispatch.dylib`_dispatch_client_callout + 8
    frame #7: 0x0000000101bb49c7 libdispatch.dylib`_dispatch_lane_serial_drain + 789
    frame #8: 0x0000000101bb587e libdispatch.dylib`_dispatch_lane_invoke + 446
    frame #9: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #10: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #11: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #205, queue = 'com.myApp.syncEngineQueue'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e1071f libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 111
    frame #3: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #4: 0x0000000101bb10b4 libdispatch.dylib`_dispatch_async_f_slow + 294
    frame #5: 0x00007fff395d53f3 CoreData`-[NSPersistentStoreCoordinator(_NSInternalMethods) managedObjectContextDidUnregisterObjectsWithIDs:generation:] + 291
    frame #6: 0x00007fff395fd513 CoreData`-[NSManagedObjectContext(_NSInternalAdditions) _disposeObjects:count:notifyParent:] + 643
    frame #7: 0x00007fff395fca76 CoreData`-[NSManagedObjectContext(_NSInternalAdditions) _dispose:] + 454
    frame #8: 0x00007fff395fc54c CoreData`-[NSManagedObjectContext _dealloc__] + 428
    frame #9: 0x00007fff395fc2ee CoreData`-[NSManagedObjectContext dealloc] + 254
    frame #10: 0x000000010018bada MyApp`-[WatcherCreate addFile:](self=0x0000600001877f40, _cmd="addFile:", filePath=@"/Users/UserName/CompanyName/MyApp/000008/1211 copy.pdf") at WatcherCreate.m:641
    frame #11: 0x0000000100183c00 MyApp`-[WatcherCreate process:](self=0x0000600001877f40, _cmd="process:", args=1 key/value pair) at WatcherCreate.m:217
    frame #12: 0x00000001001e7a3b MyApp`-[SyncWatcher handleDriverEvent:](self=0x000060000211da40, _cmd="handleDriverEvent:", event=2 key/value pairs) at SyncWatcher.m:925
    frame #13: 0x00000001001e72e1 MyApp`__37-[SyncWatcher didReceiveDriverEvent:]_block_invoke(.block_descriptor=0x0000600000ca8d80) at SyncWatcher.m:883
    frame #14: 0x00000001003be25a MyApp`thunk for @escaping @callee_unowned @convention(block) () -> () at <compiler-generated>:0
    frame #15: 0x00000001003e47cd MyApp`partial apply for thunk for @escaping @callee_unowned @convention(block) () -> () at <compiler-generated>:0
    frame #16: 0x00000001003e4667 MyApp`closure #1 in closure #1 in ConcurrentTasksQueue.add(this=0x0000600000cc00c0, task=0x00000001003e47c0 MyApp`partial apply forwarder for reabstraction thunk helper from @escaping @callee_unowned @convention(block) () -> () to @escaping @callee_guaranteed () -> () at <compiler-generated>) at ConcurrentTasksQueue.swift:38
    frame #17: 0x00000001003e5065 MyApp`partial apply for closure #1 in closure #1 in ConcurrentTasksQueue.add(task:) at <compiler-generated>:0
    frame #18: 0x00000001003bc74d MyApp`thunk for @escaping @callee_guaranteed () -> () at <compiler-generated>:0
    frame #19: 0x0000000101bab6eb libdispatch.dylib`_dispatch_call_block_and_release + 12
    frame #20: 0x0000000101bac7c3 libdispatch.dylib`_dispatch_client_callout + 8
    frame #21: 0x0000000101bafbcd libdispatch.dylib`_dispatch_continuation_pop + 576
    frame #22: 0x0000000101baeca3 libdispatch.dylib`_dispatch_async_redirect_invoke + 780
    frame #23: 0x0000000101bbf270 libdispatch.dylib`_dispatch_root_queue_drain + 334
    frame #24: 0x0000000101bbfd33 libdispatch.dylib`_dispatch_worker_thread2 + 125
    frame #25: 0x0000000101c27119 libsystem_pthread.dylib`_pthread_wqthread + 619
    frame #26: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #206, queue = 'updateStatusbarQueue'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x0000000101bad0b6 libdispatch.dylib`_dispatch_ulock_wait + 47
    frame #2: 0x0000000101bad195 libdispatch.dylib`_dispatch_thread_event_wait_slow + 33
    frame #3: 0x0000000101bbc66d libdispatch.dylib`__DISPATCH_WAIT_FOR_QUEUE__ + 309
    frame #4: 0x0000000101bbbe3f libdispatch.dylib`_dispatch_sync_f_slow + 223
    frame #5: 0x000000010033aaec MyApp`runOnMainQueue(block=0x00000001000d94d0) at PBMainThreadRun.m:18
    frame #6: 0x00000001000d8a41 MyApp`__40-[StatusBarManager updateStatusMenuItem]_block_invoke(.block_descriptor=0x0000600000cf0660) at StatusBarManager.m:277
    frame #7: 0x0000000101bab6eb libdispatch.dylib`_dispatch_call_block_and_release + 12
    frame #8: 0x0000000101bac7c3 libdispatch.dylib`_dispatch_client_callout + 8
    frame #9: 0x0000000101bb49c7 libdispatch.dylib`_dispatch_lane_serial_drain + 789
    frame #10: 0x0000000101bb587e libdispatch.dylib`_dispatch_lane_invoke + 446
    frame #11: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #12: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #13: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #207, queue = 'com.apple.CFNetwork.LoaderQ'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e1071f libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 111
    frame #3: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #4: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
    frame #5: 0x00007fff6583b1c9 libnetwork.dylib`nw_queue_async_current + 57
    frame #6: 0x00007fff6587b4f9 libnetwork.dylib`nw_endpoint_flow_protocol_disconnected + 505
    frame #7: 0x00007fff642903ac libboringssl.dylib`nw_protocol_boringssl_disconnect + 229
    frame #8: 0x00007fff6588cc80 libnetwork.dylib`-[NWConcrete_nw_endpoint_flow cancelWithHandler:forced:] + 528
    frame #9: 0x00007fff6586d628 libnetwork.dylib`nw_endpoint_handler_cancel + 1336
    frame #10: 0x00007fff6588ca5b libnetwork.dylib`__60-[NWConcrete_nw_endpoint_resolver cancelWithHandler:forced:]_block_invoke + 27
    frame #11: 0x00007fff66edb5dd libxpc.dylib`xpc_array_apply + 57
    frame #12: 0x00007fff6588c54e libnetwork.dylib`-[NWConcrete_nw_endpoint_resolver cancelWithHandler:forced:] + 238
    frame #13: 0x00007fff6586d628 libnetwork.dylib`nw_endpoint_handler_cancel + 1336
    frame #14: 0x00007fff658d9feb libnetwork.dylib`__57-[NWConcrete_nw_endpoint_proxy cancelWithHandler:forced:]_block_invoke + 27
    frame #15: 0x00007fff66edb5dd libxpc.dylib`xpc_array_apply + 57
    frame #16: 0x00007fff658d9113 libnetwork.dylib`-[NWConcrete_nw_endpoint_proxy cancelWithHandler:forced:] + 1363
    frame #17: 0x00007fff6586d628 libnetwork.dylib`nw_endpoint_handler_cancel + 1336
    frame #18: 0x00007fff65885076 libnetwork.dylib`__nw_connection_cancel_inner_block_invoke + 998
    frame #19: 0x00007fff65815307 libnetwork.dylib`nw_queue_context_async_if_needed + 87
    frame #20: 0x00007fff65884ac0 libnetwork.dylib`nw_connection_cancel_inner + 192
    frame #21: 0x00007fff65884787 libnetwork.dylib`tcp_connection_cancel + 615
    frame #22: 0x00007fff38940994 CFNetwork`TCPIOConnection::_cleanupConnection() + 294
    frame #23: 0x00007fff389740c9 CFNetwork`HTTPEngine::_cleanupConnection() + 77
    frame #24: 0x00007fff38973eaa CFNetwork`HTTPConnection::_onqueue_doNotAllowMoreRequests() + 630
    frame #25: 0x00007fff38973b4f CFNetwork`HTTPConnectionCacheEntry::ConnectionArray::stopAndRemove(long) + 99
    frame #26: 0x00007fff38973a7a CFNetwork`HTTPConnectionCacheEntry::_removeConnection(std::__1::shared_ptr<HTTPConnection>) + 162
    frame #27: 0x00007fff389737de CFNetwork`HTTPConnectionCacheEntry::purgeIdleConnections(double, double) + 482
    frame #28: 0x00007fff389733a2 CFNetwork`HTTPConnectionCache::performIdleSweep() + 182
    frame #29: 0x00007fff38973082 CFNetwork`HTTPConnectionCache::timeoutIdleConnections() + 30
    frame #30: 0x00007fff389cd3b7 CFNetwork`___ZNK18QCoreSchedulingSet12performAsyncEU13block_pointerFvvE_block_invoke + 41
    frame #31: 0x0000000101bab6eb libdispatch.dylib`_dispatch_call_block_and_release + 12
    frame #32: 0x0000000101bac7c3 libdispatch.dylib`_dispatch_client_callout + 8
    frame #33: 0x0000000101bb49c7 libdispatch.dylib`_dispatch_lane_serial_drain + 789
    frame #34: 0x0000000101bb58b1 libdispatch.dylib`_dispatch_lane_invoke + 497
    frame #35: 0x0000000101bb73e7 libdispatch.dylib`_dispatch_workloop_invoke + 3562
    frame #36: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #37: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #38: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #209, queue = 'com.apple.root.default-qos.overcommit'
    frame #0: 0x00007fff66de6a86 libsystem_kernel.dylib`__psynch_rw_wrlock + 10
    frame #1: 0x0000000101c299d2 libsystem_pthread.dylib`_pthread_rwlock_lock_wait + 89
    frame #2: 0x0000000101c2765b libsystem_pthread.dylib`_pthread_rwlock_lock_slow + 558
    frame #3: 0x0000000100e1439d libBacktraceRecording.dylib`enter_stack_in_backtrace_uniquing_table + 62
    frame #4: 0x0000000100e1092e libBacktraceRecording.dylib`gcd_queue_item_enqueue_hook + 638
    frame #5: 0x0000000101be2204 libdispatch.dylib`_dispatch_introspection_queue_item_enqueue_hook + 46
    frame #6: 0x0000000101bb1ac2 libdispatch.dylib`dispatch_async + 286
    frame #7: 0x00007fff3b0aa85d LaunchServices`LSNotificationReceiver::receiveNotificationFromServer(_xpc_connection_s*, void*) + 973
    frame #8: 0x00007fff3b0aa46c LaunchServices`___ZN22LSNotificationReceiver23getNotificationListenerEv_block_invoke.10 + 414
    frame #9: 0x00007fff66edca28 libxpc.dylib`_xpc_connection_call_event_handler + 56
    frame #10: 0x00007fff66eda9be libxpc.dylib`_xpc_connection_mach_event + 933
    frame #11: 0x0000000101bac863 libdispatch.dylib`_dispatch_client_callout4 + 9
    frame #12: 0x0000000101bc80a9 libdispatch.dylib`_dispatch_mach_msg_invoke + 487
    frame #13: 0x0000000101bb47c9 libdispatch.dylib`_dispatch_lane_serial_drain + 279
    frame #14: 0x0000000101bc93ed libdispatch.dylib`_dispatch_mach_invoke + 557
    frame #15: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #16: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #17: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
    frame #18: 0x0000000101c07600 libdispatch.dylib`_dispatch_root_queues + 896
  thread #210, queue = 'cocoa.lumberjack.ttyLogger'
    frame #0: 0x00007fff66de6a86 libsystem_kernel.dylib`__psynch_rw_wrlock + 10
    frame #1: 0x0000000101c299d2 libsystem_pthread.dylib`_pthread_rwlock_lock_wait + 89
    frame #2: 0x0000000101c2765b libsystem_pthread.dylib`_pthread_rwlock_lock_slow + 558
    frame #3: 0x0000000100e14739 libBacktraceRecording.dylib`remove_stack_from_backtrace_uniquing_table + 33
    frame #4: 0x0000000100e1147f libBacktraceRecording.dylib`decrement_work_item_refcount + 80
    frame #5: 0x0000000100e10d7b libBacktraceRecording.dylib`gcd_queue_item_complete_hook + 166
    frame #6: 0x0000000101bb4bfb libdispatch.dylib`_dispatch_lane_serial_drain + 1353
    frame #7: 0x0000000101bb587e libdispatch.dylib`_dispatch_lane_invoke + 446
    frame #8: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #9: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #10: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #212, queue = 'GCDAsyncSocket'
    frame #0: 0x00007fff66de494e libsystem_kernel.dylib`__ulock_wait + 10
    frame #1: 0x00007fff66e91c57 libsystem_platform.dylib`_os_unfair_lock_lock_slow + 140
    frame #2: 0x0000000100e10bcc libBacktraceRecording.dylib`gcd_queue_item_dequeue_hook + 219
    frame #3: 0x0000000101be23c6 libdispatch.dylib`_dispatch_introspection_queue_item_dequeue_hook + 46
    frame #4: 0x0000000101bafb35 libdispatch.dylib`_dispatch_continuation_pop + 424
    frame #5: 0x0000000101bc4506 libdispatch.dylib`_dispatch_source_invoke + 2109
    frame #6: 0x0000000101bb47c9 libdispatch.dylib`_dispatch_lane_serial_drain + 279
    frame #7: 0x0000000101bb587e libdispatch.dylib`_dispatch_lane_invoke + 446
    frame #8: 0x0000000101bc0f44 libdispatch.dylib`_dispatch_workloop_worker_thread + 691
    frame #9: 0x0000000101c27047 libsystem_pthread.dylib`_pthread_wqthread + 409
    frame #10: 0x0000000101c26e41 libsystem_pthread.dylib`start_wqthread + 13
  thread #213
    frame #0: 0x0000000101c26e34 libsystem_pthread.dylib`start_wqthread
(lldb) 

Printing description of _loggingGroup:
<OS_dispatch_group: group[0x600002104280] = { xref = 1, ref = 2, count = 1, gen = 515, waiters = 1, notifs = 0 }>

Printing description of _queueSemaphore:
<OS_dispatch_semaphore: semaphore[0x6000021042d0] = { xref = 1, ref = 1, port = 0x0, value = 999, orig = 1000 }>

I had updated my code on 8th Jan 2019.

**Comment by hhanesand** [comment]
@Speakus That will work, yes. You can verify that it is by viewing the preprocessed source by pressing the button that looks like 4 squares all the way to the left in the bar above the main editor (then select Preprocess). The call to DDLog will look like this when `LOG_ASYNC_ENABLED == NO` :

```
... [DDLog log : __objc_no level : ...
```

The first parameter `__objc_no` means sync logging.

Now, as mentioned above, we'd love for you to :
1. update to 3.5.0 to see if your issue is fixed
2. post a full backtrace generated by `bt all`

**Comment by hhanesand** [comment]
Closing this. If this issue is reproduced on version 3.5.0 or later, please open a new issue.
